src/components/ec-metroline-item/ec-metroline-item.vue

```diff
@@ -0,0 +1,203 @@
+        <div
+          data-test="ec-metroline-item__header-heading"
+          class="ec-metroline-item__header-heading"
+        >
+          <slot name="heading" />
+        </div>
+
+        <div
+          v-if="isCompleted"
+          data-test="ec-metroline-item__header-sub-heading"
+          class="ec-metroline-item__header-sub-heading"
+        >
+          <slot name="sub-heading" />
+        </div>
+
+        <div
+          v-if="isCompleted"
+          data-test="ec-metroline-item__header-cta"
+        >
+          <slot name="header-cta" />
+        </div>
+      </div>
+
+      <div
+        v-if="!isNext"
+        class="ec-metroline-item__main"
+        data-test="ec-metroline-item__main"
+      >
+        <slot name="main" />
+      </div>
+
+      <div
+        v-if="!isNext"
+        class="ec-metroline-item__footer"
+        data-test="ec-metroline-item__footer"
+      >
+        <div
+          class="ec-metroline-item__footer-cta"
+          data-test="ec-metroline-item__footer-cta"
+        >
+          <slot name="footer-cta" />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import EcIcon from '../ec-icon';
+import * as STATUS from '../../enums/metroline-status';
+
+export default {
+  name: 'EcMetrolineItem',
+  components: { EcIcon },
+  props: {
+    index: {
+      type: String,
+      required: true,
+    },
+    isLast: {
+      type: Boolean,
+      default: false,
+    },
+    status: {
+      type: String,
+      required: true,
+      validator(value) {
+        return [STATUS.NEXT, STATUS.ACTIVE, STATUS.COMPLETED].includes(value);
+      },
+    },
+  },
+  computed: {
+    isNext() {
+      return this.status === STATUS.NEXT;
+    },
+    isActive() {
+      return this.status === STATUS.ACTIVE;
+    },
+    isCompleted() {
+      return this.status === STATUS.COMPLETED;
+    },
+  },
+};
+</script>
+
+<style>
+.ec-metroline-item {
+  @apply tw-w-full;
+  @apply tw-flex;
+  @apply tw-mb-8;
+
+  &--is-last {
+    @apply tw-mb-0;
+  }
+
+  &__status {
+    @apply tw-flex tw-flex-col tw-items-center;
+    @apply tw-mr-16;
+  }
+
+  &__status-disk {
+    @apply tw-flex tw-justify-center tw-items-center;
+    @apply tw-bg-gray-6 tw-text-gray-3 tw-fill-current;
+    @apply tw-rounded-1/2;
+    @apply tw-min-w-24 tw-min-h-24;
+    @apply tw-small-text;
+    @apply tw-mb-8;
+
+    .ec-metroline-item--active & {
+      @apply tw-bg-key-4 tw-text-gray-8;
+    }
+  }
+
+  &__status-bar {
+    @apply tw-h-full tw-min-h-24;
+    @apply tw-border tw-border-solid tw-border-key-6;
+  }
+
+  &__content {
+    @apply tw-w-full;
+  }
+
+  &__header {
+    @apply tw-flex tw-flex-wrap tw-justify-between tw-items-center;
+    @apply tw-mb-12;
+  }
+
+  &__header-heading {
+    @apply tw-text-gray-3 tw-h4;
+
+    .ec-metroline-item--active & {
+      @apply tw-text-gray-3 tw-h3;
+
+      line-height: 24px; /* The disc is 24px. subsequently all header items must have a size of 24px or less. */
```

Comment:
I am aware of the design. I used a hack here without visually breaking the design to stop the header from "jumbing" when we swap from active to complete state and also keep it centered. 

IMHO I am a little concerned with centering the icon vertically as well. this was my first approach and I stumbled upon the following:

- Since the disc could move to align itself with the header now the gap between the line and the disc was varying depending which header we were showing
- This varied header height also means we still have the "jumbing" issue
- Fixed values for justifying the icon and the line together

Reply:
Have you spoken with designers about this problem? Explain them that changing the font-size/line-height of the headers are causing the content moving. To me, 24px line-height with 20px might cause an issues with readability of the text is it's on multiple lines. Don't forget that 20px font-size can generate text with 22-23px height, because of accents, that means there will be only 1-2px between the texts on multiple lines. In ideal world, line-height should be a minimum of 1.5x of the font-size.